Solutions/Azure Activity/Analytic Rules/Granting_Permissions_To_Account_detection.yaml

```diff
@@ -33,37 +33,45 @@ query: |
+  // Returns all the records from the right side that don't have matches from the left.
   | join kind = rightanti ( 
   createRoleAssignmentActivity
   | where TimeGenerated > ago(endtime)
-  | extend PrincipalId = tostring(parse_json(tostring(parse_json(tostring(Properties_d.requestbody)).Properties)).PrincipalId)
-  | extend PrincipalType = tostring(parse_json(tostring(parse_json(tostring(Properties_d.requestbody)).Properties)).PrincipalType)
-  | extend Scope = tostring(parse_json(tostring(parse_json(tostring(Properties_d.requestbody)).Properties)).Scope)
-  | extend RoleAddedDetails = tostring(parse_json(tostring(parse_json(tostring(parse_json(Properties).requestbody)).Properties)).RoleDefinitionId) 
-  | summarize StartTimeUtc = min(TimeGenerated), EndTimeUtc = max(TimeGenerated), ActivityTimeStamp = make_set(TimeGenerated), ActivityStatusValue = make_set(ActivityStatusValue), 
-  OperationIds = make_set(OperationId), CorrelationId = make_set(CorrelationId), ActivityCountByCallerIPAddress = count()  
+  | extend PrincipalId = tostring(iff(Properties_d has_cs 'PrincipalId',parse_json(tostring(parse_json(Properties_d).requestbody)).Properties.PrincipalId,parse_json(tostring(parse_json(Properties_d).requestbody)).properties.principalId))
+  | extend PrincipalType = tostring(iff(Properties_d has_cs 'PrincipalType',parse_json(tostring(parse_json(Properties_d).requestbody)).Properties.PrincipalType,parse_json(tostring(parse_json(Properties_d).requestbody)).properties.principalType))
+  | extend Scope = tostring(iff(Properties_d has_cs 'Scope',parse_json(tostring(parse_json(Properties_d).requestbody)).Properties.Scope,parse_json(tostring(parse_json(Properties_d).requestbody)).properties.scope))
+  | extend RoleAddedDetails = tostring(iff(Properties_d has_cs 'RoleDefinitionId',parse_json(tostring(parse_json(Properties_d).requestbody)).Properties.RoleDefinitionId,parse_json(tostring(parse_json(Properties_d).requestbody)).properties.roleDefinitionId))
+  | summarize StartTimeUtc = min(TimeGenerated), EndTimeUtc = max(TimeGenerated), ActivityTimeStamp = make_set(TimeGenerated), ActivityStatusValue = make_set(ActivityStatusValue), CorrelationId = make_set(CorrelationId), ActivityCountByCallerIPAddress = count()  
   by ResourceId, CallerIpAddress, Caller, OperationNameValue, Resource, ResourceGroup, PrincipalId, PrincipalType, Scope, RoleAddedDetails
   ) on CallerIpAddress, Caller
   | extend timestamp = StartTimeUtc, AccountCustomEntity = Caller, IPCustomEntity = CallerIpAddress;
   let RoleAssignedActivitywithRoleDetails = RoleAssignedActivity
   | extend RoleAssignedID = tostring(split(RoleAddedDetails, "/")[-1])
+  // Returns all matching records from left and right sides.
   | join kind = inner (AzureBuiltInRole 
   ) on $left.RoleAssignedID == $right.ID;
-  RoleAssignedActivitywithRoleDetails
+  let CountbyCallernIP = RoleAssignedActivitywithRoleDetails | summarize AssignmentCountbyCaller = count() by Caller, CallerIpAddress;
```

Comment:
This is a confusion variable name - can I suggest something more understandbale - like CallerIPCountSummary

Reply:
thanks @petebryan updated 'CountbyCallernIP' to 'CallerIPCountSummary'